### PR TITLE
CBOR correctly skips particular values

### PR DESCRIPTION
**Pull request by ardune**
The CBOR "ignoreUnknownKeys" allows for arbitrary content to be skipped.
In the case of an empty map, empty array, or tags on non-map/array values the skip would fail

Included somewhat trimmed down example test.

### CODE REVIEW

**Comment by sandwwraith** [comment]
Have you filed an issue somewhere already or you've decided to do a PR right away?

**Comment by ardune** [comment]
I just made a PR - Apologies if I skipped a step. I would be happy to open an issue or otherwise add to the PR.

File: formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Decoder.kt [hunk 1]
```diff
@@ -426,9 +426,11 @@ internal class CborParser(private val input: ByteArrayInput, private val verifyO
             } else {
                 val header = curByte and 0b111_00000
                 val length = elementLength()
-                if (header == HEADER_ARRAY || header == HEADER_MAP) {
+                if (header == HEADER_TAG) {
+                    readNumber()
+                } else if (header == HEADER_ARRAY || header == HEADER_MAP) {
```
Comment (sandwwraith) [inline formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Decoder.kt:431]: cc @JesusMcCloud as author of these changes. I do not see any particular problems, so I'll merge this if you don't have other comments.

Comment (JesusMcCloud) [reply formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Decoder.kt:431]: Thanks! Looks good to me!

**Review by sandwwraith — APPROVED** [review]
Thank you!

**Pull request merged**